src/array/cpu/coo_remove.cc

```diff
@@ -0,0 +1,48 @@
+/*!
+ *  Copyright (c) 2020 by Contributors
+ * \file array/cpu/coo_remove.cc
+ * \brief COO matrix remove entries CPU implementation
+ */
+#include <dgl/array.h>
+#include <utility>
+#include <vector>
+#include "array_utils.h"
+
+namespace dgl {
+using runtime::NDArray;
+namespace aten {
+namespace impl {
+
+template <DLDeviceType XPU, typename IdType>
+std::pair<COOMatrix, IdArray> COORemove(COOMatrix coo, IdArray entries) {
+  const int64_t nnz = coo.row->shape[0];
+  const IdType *row_data = static_cast<IdType *>(coo.row->data);
+  const IdType *col_data = static_cast<IdType *>(coo.col->data);
+  const IdType *eid_data = COOHasData(coo) ? static_cast<IdType *>(coo.data->data) : nullptr;
+
+  IdHashMap<IdType> eid_map(entries);
+
+  std::vector<IdType> new_rows, new_cols, new_eids;
+
+  for (int64_t i = 0; i < nnz; ++i) {
+    const IdType eid = eid_data ? eid_data[i] : i;
+    if (eid_map.Contains(eid))
+      continue;
+    new_rows.push_back(row_data[i]);
+    new_cols.push_back(col_data[i]);
+    new_eids.push_back(eid);
+  }
```

Comment:
In the most common case, the given `entries` are the positions to delete and the eid array is consecutive, which has a better implementation  when the entries array is small (sort entries + a linear scan). The current implementation has O(|NNZ|) number of hashmap queries, but when the entries array is small, the bloom filter in `IdHashMap` might perform good. Could you quickly test one such case to measure the gap?

Reply:
The most common case I can think of is removing the positive and negative pairs from a sampled frontier.  This is essentially done by:
```python
frontier = sample_neighbors(g, seeds)
pair_eids = g.edge_ids(heads, tails)
new_frontier = dgl.remove_edges(frontier, pair_eids)
```
So the edge IDs are not necessarily consecutive, and often times, `pair_eids` is a small array.  If we want to optimize something we should probably aim for this scenario instead.

I will still do the test though.

Reply:
Will do.  The `eid` array being consecutive should help.

Reply:
Tried the setting of link prediction on Reddit.  The edges to be removed are the positive and negative training pairs.  Sorting entries consistently outperforms hash maps by around 20%.

Varied the number of negative examples, batch size, random walk window size, number of layers and number of neighbors.